### ISSUE TITLE
FlyoutItem containing `a` isn't accessable with keyboard

### ISSUE THREAD

**Issue by christopherGdynia** [telekom/scale#915]
Hello there,

My Flyout is containing an `anchor` `tag`, but selecting it with `Space` won't trigger the `anchor` `tag`.

```tsx

import {
  ScaleButton, ScaleIconActionLink, ScaleIconActionMore, ScaleMenuFlyout, ScaleMenuFlyoutDivider, ScaleMenuFlyoutItem, ScaleMenuFlyoutList
} from '@telekom/scale-components-react';
import { forwardRef, ReactNode } from 'react';

const BaseFlyout = forwardRef<
  HTMLScaleMenuFlyoutElement,
  { url: string; name: string; children: ReactNode }
>((props, ref) => (
  <ScaleMenuFlyout ref={ref}>
    <ScaleButton slot="trigger" iconOnly variant="ghost" aria-label="Kontextmenü öffnen">
      <ScaleIconActionMore />
    </ScaleButton>
    <ScaleMenuFlyoutList>
      <ScaleMenuFlyoutItem>Portal {props.name}</ScaleMenuFlyoutItem>
      <ScaleMenuFlyoutDivider />
      {props.children}
      <ScaleMenuFlyoutItem data-link>
        <ScaleIconActionLink slot="prefix" />
        <a href={props.url} target="_blank" rel="noreferrer" className="link">
          Öffnen
        </a>
      </ScaleMenuFlyoutItem>
    </ScaleMenuFlyoutList>
  </ScaleMenuFlyout>
));

export default BaseFlyout;

```

With workaround
```tsx

import { ScaleIconActionRemove, ScaleMenuFlyoutItem } from '@telekom/scale-components-react';
import { useEventListenerFactory } from 'lotus-hooks';
import { useSelectedPortalsStore, useSetSelectedPortals } from 'src/hooks';
import BaseFlyout from '../shared/baseFlyout';
export interface IFlyoutProps {
  name: string;
  url: string;
}

const Flyout = ({ name, url }: IFlyoutProps) => {
  const selectedPortals = useSelectedPortalsStore();
  const setSelectedPortals = useSetSelectedPortals();

  const ref = useEventListenerFactory<HTMLScaleMenuFlyoutElement>(
    'scale-select' as any,
    (e: any) => {
      if (e.detail.item.getAttribute('data-remove')) {
        const newState = [...selectedPortals];
        newState.splice(newState.indexOf(name), 1);
        setSelectedPortals(newState);
      }
      if (e.detail.item.getAttribute('data-link')) {
        window.open(e.target.querySelector('a').getAttribute('href'), '_blank');
      }
    }
  );
  return (
    <>
      <BaseFlyout ref={ref} name={name} url={url}>
        <ScaleMenuFlyoutItem data-remove>
          <ScaleIconActionRemove slot="prefix" /> Vom eigenen Bereich entfernen
        </ScaleMenuFlyoutItem>
      </BaseFlyout>
    </>
  );
};

export default Flyout;

```

**Comment by acstll** [comment]
Yes, this is a limitation, links don't work as expected inside menu items. The Flyout Menu is intended to be used as a contextual menu (providing options, etc.), not for navigation. ([Here's an interesting question in stackoverflow.](https://stackoverflow.com/questions/41141247/aria-role-menuitem-for-a-or-li)) Instead of using an anchor tag, would it be possible for you to use a plain menu item, listen for the `scale-select` menu and navigate programatically (e.g. `window.location.href = '...'` or the equivalent router method)?

**Comment by acstll** [comment]
Nevertheless, I think this deserves a deeper look, also involving concept/design.

**Comment by christopherGdynia** [comment]
@acstll I already applied this work around, but this is quiet annoying in terms of accessibility ^^

For accessiblity I need to use anchor tags for links and buttons for functions. Each element needs to be accessible by keyboard. 
I applied an extra `onScaleSelect ` listener for selecting a field containing an `a` tag and setting the href. As shown above in my question.

Instead of an url change with `window.location.href` I want a new window like `targe="_blank"`, that's why I am using `window.open(..., "_blank")`

**Comment by christopherGdynia** [comment]
Hello @acstll 

I have noticed that in my application I can't open href in a new tab with `ctrl` in the head the event element contains the ctrlKey property, but scale-select doesnt


My question is can we add the ctrlKey to scale-select event

Maybe its missing in other custom events whi`ch are accessible with keyboards?

**Comment by acstll** [comment]
hey @christopherGdynia, sorry for the late reply…

We can include the original click event that has `ctrlKey` inside the details of the emitted `scale-select` event. That should be an easy fix. We could name it `originalEvent` (in a jQuery fashion). Would this work?

Also, regarding the original issue. I've very recently learned it's ok to use a link within such menus, but the `<a>` itself needs to be the `role="menuitem"`. So I'm trying to think about how to solve it, but it shouldn't be too hard either.

**Comment by christopherGdynia** [comment]
hey @acstll 

adding `originalEvent` to the `scale-events` would work perfectly.

I am just wondering, where this could be usefull aswell :D

Thanks for the informationen, I will add the `role="menuitem"` to it.

**Comment by tshimber** [comment]
seems to be fixed. @christopherGdynia please tell if the issue is still relevant
